.github/workflows/rust.yml

```diff
@@ -3,28 +3,58 @@ name: Rust
 on: [push]
 
 jobs:
-  build:
+
+  codestyle:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Rust
+        uses: hecrj/setup-rust-action@v1
+        with:
+          components: rustfmt
+          rust-version: nightly
+      - uses: actions/checkout@v2
+      - run: cargo fmt -- --check --config-path <(echo 'license_template_path = "HEADER"')
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Rust
+        uses: hecrj/setup-rust-action@v1
+        with:
+          components: clippy
+      - uses: actions/checkout@v2
+      - run: cargo clippy --all-targets -- -D clippy::all
+
+  compile:
     runs-on: ubuntu-latest
+    steps:
+      - name: Set up Rust
+        uses: hecrj/setup-rust-action@v1
+      - uses: actions/checkout@master
+      - run: cargo check --all
 
+  test:
+    needs: [codestyle, lint, compile]
```

Comment:
Why the dependency, on rustfmt in particular? I think this makes the build take a minute longer and would probably avoid trying to run the testsuite if a formatting problem was found?

Reply:
This isn't strictly required, certainly. We could remove the job dependencies in favor of speed.

Reply:
I've removed these.